### PR TITLE
[librtmp] Refactor librtmp

### PR DESCRIPTION
**Pull request by talregev**
All the credits if for @WangWeiLin-MV 
All changes taken from here #38269

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/librtmp/portfile.cmake [hunk 1]
```diff
@@ -9,21 +9,37 @@ vcpkg_from_github(
...
+        "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt"
+        "${CMAKE_CURRENT_LIST_DIR}/librtmp.def"
+    DESTINATION "${SOURCE_PATH}/librtmp"
+)
 
-vcpkg_cmake_configure(
-    SOURCE_PATH "${SOURCE_PATH}"
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        crypto LIBRTMP_CRYPTO
```
Comment (LilyWangLL) [inline ports/librtmp/portfile.cmake:23]:
```suggestion
```
Please also remove the following `)`.

File: ports/librtmp/portfile.cmake [hunk 2]
```diff
@@ -9,21 +9,37 @@ vcpkg_from_github(
...
 )
+if(LIBRTMP_CRYPTO)
```
Comment (LilyWangLL) [inline ports/librtmp/portfile.cmake:25]:
```suggestion
if("crypto" IN_LIST FEATURES)
```

File: ports/librtmp/portfile.cmake [hunk 3]
```diff
@@ -9,21 +9,37 @@ vcpkg_from_github(
...
+    list(APPEND FEATURE_OPTIONS "-DLIBRTMP_SSL=OPENSSL")
+endif()
 
-vcpkg_cmake_install()
+include(CMakePrintHelpers)
+cmake_print_variables(FEATURE_OPTIONS)
```
Comment (LilyWangLL) [inline ports/librtmp/portfile.cmake:30]:
```suggestion
```

File: ports/librtmp/portfile.cmake [hunk 4]
```diff
@@ -9,21 +9,37 @@ vcpkg_from_github(
...
+vcpkg_cmake_configure(SOURCE_PATH "${SOURCE_PATH}/librtmp"
```
Comment (LilyWangLL) [inline ports/librtmp/portfile.cmake:31]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}/librtmp"
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by talregev** [comment]
It ready for review.

**Review by LilyWangLL — DISMISSED** [review]

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

**Comment by dg0yt** [comment]
Has anybody the servers and apps to test this at runtime, in particular with openssl 3?

---

Before adding review comments, I want to know if the vcpkg maintainers want to consider this PR for merging:

https://github.com/microsoft/vcpkg/pull/38161#issuecomment-2166826988 recorded a broad opinion of "we should not change it until upstream makes an actual release." It is still unclear if there will be such a release, and if there would be a tarball for it.
http://lists.mplayerhq.hu/pipermail/rtmpdump/2024-June/002561.html

#38161 was closed with the option for "reopening this given an actual release by upstream", https://github.com/microsoft/vcpkg/pull/38161#issuecomment-2166826988. 

Upstream clarified that nobody should be using (the protocol) RTMP any more:

> Adobe officially killed Flash, as of December 31, 2020.
> 
> https://www.howtogeek.com/700229/adobe-flash-is-deadheres-what-that-means/
> 
> Flash was the only official client for RTMP. With that gone, there's no reason for anybody to
be using an RTMP server any more, and thus no reason for anyone to be using librtmp in any
clients any more.

http://lists.mplayerhq.hu/pipermail/rtmpdump/2024-June/002561.html

File: ports/librtmp/portfile.cmake (hunk 1 above)
Comment (talregev) [reply ports/librtmp/portfile.cmake]: I will restore.

**Comment by talregev** [comment]
@LilyWangLL I restore the changes.
Please review again. Thank you!

**Comment by LilyWangLL** [comment]
Pinging @BillyONeal for response, before there are new updates upstream, do we continue to maintain this port?

**Comment by BillyONeal** [comment]
@dg0yt 's statement above makes a lot of sense to me. If upstream's opinion is that nobody should use this any longer and that's why they aren't making a release, I'm a bit nervous to touch anything about the port anymore as its days are numbered and any change results in nonzero chance of breakage...

**Comment by talregev** [comment]
When upstream's opinion is that nobody should use this any longer, but still update the git and give it hint on the commit about 2.6 for me it feels strange.
@dg0yt @BillyONeal Your concerns is valid.
I think I will close this PR.

**Pull request closed without merging**